src/search/html.ml

```diff
@@ -3,11 +3,15 @@ type html = Html_types.div_content Tyxml.Html.elt
 open Odoc_model
 open Lang
 
-let url id =
-  match
-    Odoc_document.Url.from_identifier ~stop_before:false
-      (id :> Odoc_model.Paths.Identifier.t)
-  with
+let url { Entry.id; kind; doc = _ } =
+  let open Entry in
+  let stop_before =
+    (* Docstring do not have an ID, so we use the ID from the parent signature
+       in search entries. Links to signature need [stop_before] to be false
+       (otherwise the link may point to the definition point of the module). *)
+    match kind with Doc _ -> false | _ -> true
```

Comment:
Hmm.

I think my point was that the intent of this line of code is to prevent a link potentially going to a module/module type/class/etc that doesn't have an expansion. I was mostly expecting to see something more like `match kind with | Module _ -> true | ModuleType _ -> true ... | _ -> false` - where the wildcard is matching the things like `Exception` and so on. I don't think it's terribly obvious that `stop_before` has no impact on those things, so instead we can just match `Doc` and let `stop_before` do nothing for the other things.